spec/ics-007-tendermint-client/README.md

```diff
@@ -200,18 +249,48 @@ function checkMisbehaviourAndUpdateState(
 }
 ```
 
+### Upgrades
+
+The chain which this light client is tracking can elect to write a special pre-determined key in state to allow the light client to update its client state (e.g. with a new chain ID or epoch) in preparation for an upgrade.
+
+As the client state change will be performed immediately, once the new client state information is written to the predetermined key, the client will no longer be able to follow blocks on the old chain, so it must upgrade promptly.
+
+```typescript
+function upgradeClientState(
+  clientState: ClientState,
+  newClientState: ClientState,
+  height: Height,
+  proof: CommitmentPrefix) {
+    // check proof of updated client state in state at predetermined commitment prefix and key
```

Comment:
How would we know what the updated client state is going to be before the upgrade, since we don't know the first header.

This could be fixed with the comment here: https://github.com/cosmos/cosmos-sdk/issues/6736#issuecomment-659374828

Tho we still wouldn't know something like `LatestTimestamp uint64` beforehand

Reply:
See below comment.